ui/apps/dashboard/src/components/Onboarding/InvokeFn.tsx

```diff
@@ -1,31 +1,101 @@
-import { useState } from 'react';
+import { useEffect, useState } from 'react';
 import { useRouter } from 'next/navigation';
+import { Alert } from '@inngest/components/Alert/Alert';
 import { NewButton } from '@inngest/components/Button';
 import { CodeBlock } from '@inngest/components/CodeBlock/CodeBlock';
+import { parseCode } from '@inngest/components/InvokeButton/utils';
 import { NewLink } from '@inngest/components/Link';
-import { Select } from '@inngest/components/Select/Select';
+import { Select, type Option } from '@inngest/components/Select/Select';
+import { toast } from 'sonner';
 
 import { pathCreator } from '@/utils/urls';
 import { OnboardingSteps } from '../Onboarding/types';
+import { invokeFunction, prefetchFunctions } from './actions';
 import useOnboardingStep from './useOnboardingStep';
 
-/* TODO: fetch list of functions action */
-const functions = [
+const initialCode = JSON.stringify(
   {
-    id: '1',
-    name: 'Function 1',
+    data: {
+      example: 'type a JSON payload here to test your function',
+    },
   },
-  {
-    id: '2',
-    name: 'Function 2',
-  },
-];
+  null,
+  2
+);
+
+interface FunctionOption extends Option {
+  slug: string;
+  current: {
+    triggers: {
+      eventName?: string;
+    }[];
+  };
+}
 
 export default function InvokeFn() {
   const { updateLastCompletedStep } = useOnboardingStep();
-  const [selectedFunction, setSelectedFunction] = useState(functions[0]);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState<string>();
+  const [functions, setFunctions] = useState<FunctionOption[]>([]);
+  const [selectedFunction, setSelectedFunction] = useState<FunctionOption | null>(null);
+  const [rawPayload, setRawPayload] = useState(initialCode);
+  const [isFnInvoked, setIsFnInvoked] = useState(false);
   const router = useRouter();
 
+  const hasEventTrigger =
+    selectedFunction?.current.triggers.some((trigger) => trigger.eventName) ?? false;
+
+  useEffect(() => {
+    const loadFunctions = async () => {
+      try {
+        setLoading(true);
+        const fetchedFunctions = await prefetchFunctions();
+        setFunctions(fetchedFunctions);
+      } catch (err) {
```

Comment:
should probably also `console.error` out here just in case we ever need to troubleshoot an issue here.

Reply:
Sure. The error is already displayed to the user in the UI, but I can add a console just in case